README.md
# WPDistillery

![WPDistillery Logo](http://files.flurinduerst.ch/wpdistillery/wpdistillery_logo.png)

Version 2.3.6 (05.11.2018)

Created by [wpdistillery.org](https://wpdistillery.org)

## What is WPDistillery

WP Distillery does all the work for you when setting up a new WordPress project with [Scotch Box](https://box.scotch.io/). Simply add your preferred theme, plugins, and options into `config.yml` and you're good to go! With WPDistillery it won't take longer than 5 minutes until you can start working on your new WordPress project.

One simple command will:

- Install Scotch Box
- Install and update requirements on the local webserver
- Download, install, and configure WordPress
- Set WordPress options
- Install and activate your favorite WordPress theme (default: [WPSeed](https://wpseed.org))
- Remove default themes
- Install and activate the plugins defined in the config files
- Clean the contents, plugins, unused files, and other Wordpress defaults

You're able to adjust which of the above tasks will be executed. Simply set the desired tasks to `true` or `false` in the **Setup Options** section at the bottom of `config.yml`. All you need to do after customizing your config file is `vagrant up`. That's it! With that one single command WPDistillery will:

1. SSH to the VM
1. Update WP-CLI
1. Install and configure everything

It also comes with an optional auto-update function and integrated support for Windows. Check out the [Changelog](CHANGELOG.md) for a complete list of changes.

WPDistillery is fully compatible with [Scotchbox 3.0](https://box.scotch.io/)

## Demo

[![WPDistillery Setup Video](http://files.flurinduerst.ch/wpdistillery/demovideo_thumb2.png)](https://youtu.be/y1GtIiODsxM)

## Dependencies

- [ScotchBox](https://box.scotch.io) (using [Vagrant](https://vagrantup.com) & [Virtualbox](https://virtualbox.org))
- [Vagrant Hostsupdater](https://github.com/cogitatio/vagrant-hostsupdater) (`vagrant plugin install vagrant-hostsupdater`)

## Setup

To setup a new project running Scotch Box and WordPress, simply follow these steps:

1. In a terminal, run: `git clone https://github.com/flurinduerst/WPDistillery.git my-project`
2. Customize `wpdistillery/config.yml` to suit your needs. See [configuration file documentation](README_CONFIG.md).
3. Inside the `Vagrantfile` add your local URL at `config.vm.hostname`. This should be the same as `wpsettings:url:` in `config.yml`.
4. Run `vagrant up` from where your `Vagrantfile` is.

Done! You can now access your project at the local URL (for example `yoursite.vm`) defined in **Step 3**. (or at http://192.168.33.10/)

### Windows Users

Windows users may wish to `vagrant plugin install vagrant-winnfsd` before `vagrant up` for a moderate speed increase on sites with a lot of files. WPDistillery will detect **WinNFSd** support if needed. See [Vagrant WinNFSd](https://github.com/winnfsd/vagrant-winnfsd) for more information.

## Additional Information

### WPDistillery with WP-Multisite

Using this [Pull Request](https://github.com/flurinduerst/WPDistillery/pull/45) you can add multisite capability for your project. We [decided](https://github.com/flurinduerst/WPDistillery/issues/59) to not merge the PR to keep WPDistillery as clean as it is.

### Auto Update WordPress and Plugins

If you want to automatically update WordPress and all Plugins on every `vagrant up` you can remove the comment character at line 26 inside the Vagrantfile.

### Windows Support

Using Windows? No Problem! WPDistillery will detect if you're using Windows and if so, automatically convert all files using dos2unix.

### Basic Vagrant Commands

- `vagrant up` will start the machine. The first ever `vagrant up` in your project will also install Scotch Box and execute provisioning.
- `vagrant provision` will execute provisioning. This is where WPDistillery runs its core function which is installing and configuring WordPress according to `config.yml`. Before that, it will also update WP-CLI and set the upload size to 64MB. Normally `vagrant provision` should not be executed manually but can be used to re-run the WPDistillery setup in case you want to re-install WordPress.
- `vagrant reload` will restart Vagrant. This is required for changes made in the `Vagrantfile` to take effect.
- `vagrant halt` will shut down the virtual machine.
- `vagrant destroy` will destroy all the resources related to the current virtual machine. This action is not reversable.

 More informations can be found at [vagrantup.com](https://vagrantup.com).

## Troubleshooting

Something went wrong within the WPDistillery setup. I'd like to restart the setup:

- Fix your setting (probably in `wpdistillery/config.yml` or `Vagrantfile`)
- Remove all tables from the `scotchbox` database
- Remove the `public` folder
- Run `vagrant reload --provision`

## About

- Author: [Flurin Dürst](https://github.com/flurinduerst) | [Twitter](https://twitter.com/flurinduerst)
- Contributors:
  - [@ShaneShipston](https://github.com/ShaneShipston)
  - [@drawcard](https://github.com/drawcard)
  - [@thursby](https://github.com/thursby)
  - [@tgroff](https://github.com/tgroff)
  - [@mohnjatthews](https://github.com/mohnjatthews)

### Contribute

- Fork it
- Create your feature branch
- Commit your changes
- Push to the branch
- Create new a Pull Request

Feel free to contact me if you have questions or need any advice.

### License

WPDistillery is released under the MIT Public License.

Note: The "About" section in `README.md` and the author (`@author`) notice in the file-headers shall not be edited or deleted without permission. For details, please see [License](LICENSE).

I’m putting a lot of time into maintaining WPDistillery, so please consider [donating](https://www.paypal.me/FlurinDuerst/10) or [sharing](https://twitter.com/intent/tweet?url=https%3A%2F%2Fwpdistillery.org). Thank you!


CHANGELOG.md

# CHANGELOG

## HEAD | 2.3.6 | 05.11.2018
* Add WP-CLI environment variables to fix permission errors when creating the WPCLI cache folder (thanks to [@tgroff](https://github.com/tgroff))
* Clean up the readme (thanks to [@mohnjatthews](https://github.com/mohnjatthews)
* fix a bug regarding the Themes mapping name in the provisioning (thanks to [@bzmr](https://github.com/bzmr))
* add WinNFSd support (thanks to [@thursby](https://github.com/thursby))
* fix license-link in documentation (thanks to [@mikaeljorhult](https://github.com/mikaeljorhult))
* update default plugin list
* fix a bug where the `public` dir was not created on setup (thanks to [@donaldCull](https://github.com/donaldCull))
* remove custom root dir because path to the docroot is in the Apache config using `public` (thanks to [@XedinUnknown](https://github.com/XedinUnknown))

Contributors: 
* [@tgroff](https://github.com/tgroff)
* [@mohnjatthews](https://github.com/mohnjatthews)

## 2.3.0 | 24.01.2018
* re-organize the theme-cleanup (now located at `themes`, removed from `cleanup`)
* fix a bug caused by using a default theme with theme-cleanup activated
* silence curl command for downloading the current version of WP CLI to reduce cli-bloating
* change default local-domain to `.vm` (for "virtual machine") as `.local` has conflicts when using browsersync
* add note about the WP-Multisite feature to the readme (thanks to [@pjohanneson](https://github.com/pjohanneson))

## 2.2.0 | 31.10.2017
* change default local-domain to `.local` as `.dev` is an official gTLD now
* add ssh credentials to `Vagrantfile`
* add `--hard` to rewrite flush to make sure .htaccess file is generated (thanks to [@XedinUnknown](https://github.com/XedinUnknown))
* Fix Typo

## 2.1.0 | 21.06.2017
* update setup for Scotch Box 3.0 (re-install WP-CLI for php7)
* update default plugin list: remove yoast, add the-seo-framework (see https://roots.io/weve-migrated-from-yoast-seo/)
* remove max-filesize option on provisioning
* add license note
* fix yaml syntax errors (thanks to [@XedinUnknown](https://github.com/XedinUnknown))
* fix front_page_name variable in setup

## 2.0.0 | 15.01.2017
* simplified setup process to `vagrant up`
* restructured file and folder structure (moved all files into a wpdistillery folder)
* added custom Vagrantfile
* added auto-update for WP-CLI
* added auto-update option for WordPress Core and Plugins
* fixed a few minor bugs
* added Windows Support

Special thanks to [@ShaneShipston](https://github.com/ShaneShipston) for his feature-suggestions and everyone involved in beta testing ([@MarcWieland95](https://github.com/marcwieland95), Anthony Chapman)

Contributors: [@FlurinDuerst](https://github.com/flurinduerst), [@ShaneShipston](https://github.com/ShaneShipston)

## 1.8.0 | 20.12.2016
* fixed a bug that occurred when sending the admin email after core installation
* fixed a bug that occurred on installing a theme from Github using the current version of WP-CLI (0.25.0)
* added an option to define WordPress Version
* documentation updates
* added rewrite flush to make sure permalinks are working correctly
* added firstname/lastname to default user
* removed google-sitemap-generator from default plugins (Yoast covers this aswell)
* added note about wpdistillery.org

Contributors: [@flurinduerst](https://github.com/flurinduerst), [@ShaneShipston](https://github.com/ShaneShipston), [@drawcard](https://github.com/drawcard)

## 1.7.1 | 02.08.2016
* fixed a bug that caused an error if a theme was installed that doesn't end with `master.zip`
* add additional options
  * custom page to be used as front_page
  * timezone
  * description
  * default image sizes
* add settings setup task
* small bugfixes
* cleanup/renaming

## 1.6.1 | 07.07.2016
* bugfix (now correctly checking `theme_rename`)
* add error handler that asks the user to continue if an error occured
* add google-sitemap-generator to preset plugins
* remove dublicated plugin entry
* remove WPDistillery Version from config/setup - remain at README

## 1.5.1 | 21.06.2016
* Rename to WP Distillery
* Bugfixes / Typo / Documentation

## 1.4.0 | 17.06.2016
* Add support for custom WP root folder
* Add option to rename theme after installation
* Add file versions to config.yml and setup.sh
* Fix Typo in setup procedure


## 1.3.0 | 16.06.2016
* Add modular setup task options

## 1.2.0 | 17.06.2016
* Public Release


setup.sh
#!/usr/bin/env bash
#
# WP SETUP FILE
#
# Author: Flurin Dürst » github.com/flurinduerst
# URL: https://github.com/flurinduerst/WPDistillery
#
# File version 1.6.3

# ERROR Handler
# ask user to continue on error
function continue_error {
  read -p "$(echo -e "${RED}Do you want to continue anyway? (y/n) ${NC}")" -n 1 -r
  if [[ ! $REPLY =~ ^[Yy]$ ]]; then
    printf "\n${RED}»»» aborting WPDistillery setup! ${NC}\n"
    exit 1
  else
    printf "\n${GRN}»»» continuing WPDistillery setup... ${NC}\n"
  fi
}
trap 'continue_error' ERR

# REQUIREMENTS
####################################################################################################

# YAML PARSER FUNCTION
function parse_yaml() {
    local prefix=$2
    local s
    local w
    local fs
    s='[[:space:]]*'
    w='[a-zA-Z0-9_]*'
    fs="$(echo @|tr @ '\034')"
    sed -ne "s|^\($s\)\($w\)$s:$s\"\(.*\)\"$s\$|\1$fs\2$fs\3|p" \
        -e "s|^\($s\)\($w\)$s[:-]$s\(.*\)$s\$|\1$fs\2$fs\3|p" "$1" |
    awk -F"$fs" '{
    indent = length($1)/2;
    vname[indent] = $2;
    for (i in vname) {if (i > indent) {delete vname[i]}}
        if (length($3) > 0) {
            vn=""; for (i=0; i<indent; i++) {vn=(vn)(vname[i])("_")}
            printf("%s%s%s=(\"%s\")\n", "'"$prefix"'",vn, $2, $3);
        }
    }' | sed 's/_=/+=/g'
}

# DEFINE COLORS
RED='\033[0;31m' # error
GRN='\033[0;32m' # success
BLU='\033[0;34m' # task
BRN='\033[0;33m' # headline
NC='\033[0m' # no color

# EXECUTIVE SETUP
####################################################################################################

printf "${BRN}========== WPDISTILLERY START ==========${NC}\n\n"

# READ CONFIG
eval $(parse_yaml config.yml "CONF_")

# CHECK WP FOLDER
if [ ! -d "$CONF_wpfolder" ]; then
  mkdir $CONF_wpfolder
  printf "${BLU}»»» creating WP Folder $CONF_wpfolder...${NC}\n"
fi

cd $CONF_wpfolder

# INSTALL WORDPRESS
if $CONF_setup_wp ; then
  printf "${BRN}[=== INSTALL WORDPRESS ===]${NC}\n"
  printf "${BLU}»»» downloading WordPress...${NC}\n"
  wp core download --locale=$CONF_wplocale --version=$CONF_wpversion
  printf "${BLU}»»» creating wp-config...${NC}\n"
  wp core config --dbname=$CONF_db_name --dbuser=$CONF_db_user --dbpass=$CONF_db_pass --dbprefix=$CONF_db_prefix --locale=$CONF_wplocale
  printf "${BLU}»»» installing wordpress...${NC}\n"
  wp core install --url=$CONF_wpsettings_url --title="$CONF_wpsettings_title" --admin_user=$CONF_admin_user --admin_password=$CONF_admin_password --admin_email=$CONF_admin_email --skip-email
  wp user update 1 --first_name=$CONF_admin_first_name --last_name=$CONF_admin_last_name
else
  printf "${BLU}>>> skipping WordPress installation...${NC}\n"
fi

if $CONF_setup_settings ; then
  printf "${BLU}»»» configure settings...${NC}\n"
  printf "» timezone:\n"
  wp option update timezone $CONF_timezone
  wp option update timezone_string $CONF_timezone
  printf "» permalink structure:\n"
  wp rewrite structure "$CONF_wpsettings_permalink_structure"
  wp rewrite flush
  printf "» description:\n"
  wp option update blogdescription "$CONF_wpsettings_description"
  printf "» image sizes:\n"
  wp option update thumbnail_size_w $CONF_wpsettings_thumbnail_width
  wp option update thumbnail_size_h $CONF_wpsettings_thumbnail_height
  wp option update medium_size_w $CONF_wpsettings_medium_width
  wp option update medium_size_h $CONF_wpsettings_medium_height
  wp option update large_size_w $CONF_wpsettings_large_width
  wp option update large_size_h $CONF_wpsettings_large_height
  if ! $CONF_wpsettings_convert_smilies ; then
    wp option update convert_smilies 0
  fi
  if $CONF_wpsettings_page_on_front ; then
    printf "» front page:\n"
    # create and set frontpage
    wp post create --post_type=page --post_title="$CONF_wpsettings_page_on_front_frontpage_name" --post_content='Front Page created by WPDistillery' --post_status=publish
    wp option update page_on_front $(wp post list --post_type=page --post_status=publish --posts_per_page=1 --pagename="$CONF_wpsettings_page_on_front_frontpage_name" --field=ID --format=ids)
    wp option update show_on_front 'page'
  fi
else
  printf "${BLU}>>> skipping settings...${NC}\n"
fi

# INSTALL THEME
if $CONF_setup_theme ; then
  printf "${BRN}[=== INSTALL $CONF_theme_name ===]${NC}\n"
  printf "${BLU}»»» downloading $CONF_theme_name...${NC}\n"
  wp theme install $CONF_theme_url
  printf "${BLU}»»» installing/activating $CONF_theme_name...${NC}\n"
  if [ ! -z "$CONF_theme_rename" ]; then
    # rename theme
    printf "${BLU}»»» renaming $CONF_theme_name to $CONF_theme_rename...${NC}\n"
    mv wp-content/themes/$CONF_theme_name wp-content/themes/$CONF_theme_rename
    wp theme activate $CONF_theme_rename
  else
    wp theme activate $CONF_theme_name
  fi
else
  printf "${BLU}>>> skipping theme installation...${NC}\n"
fi

# CLEANUP
if $CONF_setup_cleanup ; then
  printf "${BRN}[=== CLEANUP ===]${NC}\n"
  if $CONF_setup_cleanup_comment ; then
    printf "${BLU}»»» removing default comment...${NC}\n"
    wp comment delete 1 --force
  fi
  if $CONF_setup_cleanup_posts ; then
    printf "${BLU}»»» removing default posts...${NC}\n"
    wp post delete 1 2 --force
  fi
  if $CONF_setup_cleanup_files ; then
    printf "${BLU}»»» removing WP readme/license files...${NC}\n"
    # delete default files
    if [ -f readme.html ];    then rm readme.html;    fi
    if [ -f license.txt ];    then rm license.txt;    fi
    # delete german files
    if [ -f liesmich.html ];  then rm liesmich.html;  fi
  fi
  if $CONF_setup_cleanup_themes ; then
    printf "${BLU}»»» removing default themes...${NC}\n"
    wp theme delete twentyfourteen
    wp theme delete twentyfifteen
    wp theme delete twentysixteen
  fi
else
  printf "${BLU}>>> skipping Cleanup...${NC}\n"
fi

# PLUGINS
if $CONF_setup_plugins ; then
  printf "${BRN}[=== PLUGINS ===]${NC}\n"
  printf "${BLU}»»» removing WP default plugins${NC}\n"
  wp plugin delete akismet
  wp plugin delete hello
  printf "${BLU}»»» adding active plugins${NC}\n"
  for entry in "${CONF_plugins_active[@]}"
  do
  	wp plugin install $entry --activate
  done

  printf "${BLU}»»» adding inactive plugins${NC}\n"
  for entry in "${CONF_plugins_inactive[@]}"
  do
    wp plugin install $entry
  done
else
  printf "${BLU}>>> skipping Plugin installation...${NC}\n"
fi

# MISC
printf "${BLU}»»» checking wp cli version${NC}\n"
wp cli check-update

printf "${BRN}========== WPDISTILLERY FINISHED ==========${NC}\n"


wpdistillery/wpdistillery.sh
#!/usr/bin/env bash
#
# WPDistillery setup file
#
# Author: Flurin Dürst
# URL: https://wpdistillery.org
#
# File version 1.8.3

# ERROR Handler
# ask user to continue on error
function continue_error {
  read -p "$(echo -e "${RED}Do you want to continue anyway? (y/n) ${NC}")" -n 1 -r
  if [[ ! $REPLY =~ ^[Yy]$ ]]; then
    printf "\n${RED}»»» aborting WPDistillery setup! ${NC}\n"
    exit 1
  else
    printf "\n${GRN}»»» continuing WPDistillery setup... ${NC}\n"
  fi
}
trap 'continue_error' ERR

# WP-CLI environment variable
export WP_CLI_CACHE_DIR=/home/vagrant/.wp-cli/cache

# REQUIREMENTS
####################################################################################################

# YAML PARSER FUNCTION
function parse_yaml() {
    local prefix=$2
    local s
    local w
    local fs
    s='[[:space:]]*'
    w='[a-zA-Z0-9_]*'
    fs="$(echo @|tr @ '\034')"
    sed -ne "s|^\($s\)\($w\)$s:$s\"\(.*\)\"$s\$|\1$fs\2$fs\3|p" \
        -e "s|^\($s\)\($w\)$s[:-]$s\(.*\)$s\$|\1$fs\2$fs\3|p" "$1" |
    awk -F"$fs" '{
    indent = length($1)/2;
    vname[indent] = $2;
    for (i in vname) {if (i > indent) {delete vname[i]}}
        if (length($3) > 0) {
            vn=""; for (i=0; i<indent; i++) {vn=(vn)(vname[i])("_")}
            printf("%s%s%s=(\"%s\")\n", "'"$prefix"'",vn, $2, $3);
        }
    }' | sed 's/_=/+=/g'
}

# DEFINE COLORS
RED='\033[0;31m' # error
GRN='\033[0;32m' # success
BLU='\033[0;34m' # task
BRN='\033[0;33m' # headline
NC='\033[0m' # no color

# EXECUTIVE SETUP
####################################################################################################

printf "${BRN}========== WPDISTILLERY START ==========${NC}\n\n"

# MOVE TO SYNCED DIRECTORY
cd ..

# READ CONFIG
eval $(parse_yaml wpdistillery/config.yml "CONF_")


# CHECK FOR ROOT-DIR
if [ ! -d "public" ]; then
  printf "${BLU}»»» creating root dir \"public\"...${NC}\n"
  mkdir public
fi

# navigate to root dir
cd public

# INSTALL WORDPRESS
if $CONF_setup_wp ; then
  printf "${BRN}[=== INSTALL WORDPRESS ===]${NC}\n"
  printf "${BLU}»»» downloading WordPress...${NC}\n"
  wp core download --locale=$CONF_wplocale --version=$CONF_wpversion
  printf "${BLU}»»» creating wp-config...${NC}\n"
  wp core config --dbname=$CONF_db_name --dbuser=$CONF_db_user --dbpass=$CONF_db_pass --dbprefix=$CONF_db_prefix --locale=$CONF_wplocale
  printf "${BLU}»»» installing WordPress...${NC}\n"
  wp core install --url=$CONF_wpsettings_url --title="$CONF_wpsettings_title" --admin_user=$CONF_admin_user --admin_password=$CONF_admin_password --admin_email=$CONF_admin_email --skip-email
  wp user update 1 --first_name=$CONF_admin_first_name --last_name=$CONF_admin_last_name
else
  printf "${BLU}>>> skipping WordPress installation...${NC}\n"
fi

if $CONF_setup_settings ; then
  printf "${BLU}»»» configure settings...${NC}\n"
  printf "» timezone:\n"
  wp option update timezone $CONF_timezone
  wp option update timezone_string $CONF_timezone
  printf "» permalink structure:\n"
  wp rewrite structure "$CONF_wpsettings_permalink_structure"
  wp rewrite flush --hard
  printf "» description:\n"
  wp option update blogdescription "$CONF_wpsettings_description"
  printf "» image sizes:\n"
  wp option update thumbnail_size_w $CONF_wpsettings_thumbnail_width
  wp option update thumbnail_size_h $CONF_wpsettings_thumbnail_height
  wp option update medium_size_w $CONF_wpsettings_medium_width
  wp option update medium_size_h $CONF_wpsettings_medium_height
  wp option update large_size_w $CONF_wpsettings_large_width
  wp option update large_size_h $CONF_wpsettings_large_height
  if ! $CONF_wpsettings_convert_smilies ; then
    wp option update convert_smilies 0
  fi
  if $CONF_wpsettings_page_on_front ; then
    printf "» front page:\n"
    # create and set frontpage
    wp post create --post_type=page --post_title="$CONF_wpsettings_frontpage_name" --post_content='Front Page created by WPDistillery' --post_status=publish
    wp option update page_on_front $(wp post list --post_type=page --post_status=publish --posts_per_page=1 --pagename="$CONF_wpsettings_frontpage_name" --field=ID --format=ids)
    wp option update show_on_front 'page'
  fi
else
  printf "${BLU}>>> skipping settings...${NC}\n"
fi

# INSTALL/REMOVE THEME
if $CONF_setup_theme ; then
  printf "${BRN}[=== CONFIGURE THEME ===]${NC}\n"
  printf "${BLU}»»» downloading $CONF_theme_name...${NC}\n"
  wp theme install $CONF_theme_url --force
  printf "${BLU}»»» installing/activating $CONF_theme_name...${NC}\n"
  if [ ! -z "$CONF_theme_rename" ]; then
    # rename theme
    printf "${BLU}»»» renaming $CONF_theme_name to $CONF_theme_rename...${NC}\n"
    mv wp-content/themes/$CONF_theme_name wp-content/themes/$CONF_theme_rename
    wp theme activate $CONF_theme_rename
  else
    wp theme activate $CONF_theme_name
  fi
  if [ ! -z "$CONF_theme_remove" ]; then
    printf "${BLU}»»» removing default themes...${NC}\n"
    # loop trough themes that shall be removed
    for loopedtheme in "${CONF_theme_remove[@]}"
    do :
      #make sure the theme to delete is not the chosen one
      if [ $loopedtheme != $CONF_theme_name ]; then
        printf "${BLU}» removing $loopedtheme...${NC}\n"
        wp theme delete $loopedtheme

      fi
    done
    # end loop
  fi

else
  printf "${BLU}>>> skipping theme installation...${NC}\n"
fi

# CLEANUP
if $CONF_setup_cleanup ; then
  printf "${BRN}[=== CLEANUP ===]${NC}\n"
  if $CONF_setup_cleanup_comment ; then
    printf "${BLU}»»» removing default comment...${NC}\n"
    wp comment delete 1 --force
  fi
  if $CONF_setup_cleanup_posts ; then
    printf "${BLU}»»» removing default posts...${NC}\n"
    wp post delete 1 2 --force
  fi
  if $CONF_setup_cleanup_files ; then
    printf "${BLU}»»» removing WP readme/license files...${NC}\n"
    # delete default files
    if [ -f readme.html ];    then rm readme.html;    fi
    if [ -f license.txt ];    then rm license.txt;    fi
    # delete german files
    if [ -f liesmich.html ];  then rm liesmich.html;  fi
  fi
else
  printf "${BLU}>>> skipping Cleanup...${NC}\n"
fi

# PLUGINS
if $CONF_setup_plugins ; then
  printf "${BRN}[=== PLUGINS ===]${NC}\n"
  printf "${BLU}»»» removing WP default plugins${NC}\n"
  wp plugin delete akismet
  wp plugin delete hello
  printf "${BLU}»»» adding active plugins${NC}\n"
  for entry in "${CONF_plugins_active[@]}"
  do
  	wp plugin install $entry --activate
  done

  printf "${BLU}»»» adding inactive plugins${NC}\n"
  for entry in "${CONF_plugins_inactive[@]}"
  do
    wp plugin install $entry
  done
else
  printf "${BLU}>>> skipping Plugin installation...${NC}\n"
fi

# MISC
printf "${BLU}»»» checking wp cli version${NC}\n"
wp cli check-update

printf "${BRN}========== WPDISTILLERY FINISHED ==========${NC}\n"
